libtransmission/torrent.cc

```diff
@@ -111,6 +111,30 @@ tr_torrent* tr_torrentFindFromObfuscatedHash(tr_session* session, tr_sha1_digest
     return nullptr;
 }
 
+bool tr_torrentSetMetainfoFromFileIfMagnet(tr_torrent* tor, tr_torrent_metainfo* metainfo, char const* filename)
+{
+    if (tr_torrentHasMetadata(tor))
+    {
+        return false;
+    }
+    tr_error* error = nullptr;
+
+    tr_torrentUseMetainfoFromFile(tor, metainfo, filename, &error);
+
+    if (error != nullptr)
+    {
+        tor->setLocalError(fmt::format(
+            _("Couldn't copy '{path_in}' to '{path_out}': {error} ({error_code})"),
+            fmt::arg("path_in", filename),
+            fmt::arg("path_out", tor->torrentFile()),
+            fmt::arg("error", error->message),
+            fmt::arg("error_code", error->code)));
```

Comment:
It would be more flexible to copy the `error->message` here instead of assuming that the error came from copying the file. At some future point `tr_torrentUseMetainfoFromFile()` may have other reasons to fail